### ISSUE TITLE
llama reproduce issue

### ISSUE THREAD

**Issue by jordane95** [allenai/open-instruct#179]
Hi,

I noticed that tuluv1 did an ablation of different instruction datasets, including flanv2, which is used in the original llama paper to train an instruct model. However, I find that the MMLU 5shot result of 65b llama model after instruct in your tuluv1 paper cannot match the number reported in the llama paper. It reported a 5 point improvement, but Table 8 in tuluv1 paper observed nearly no if not negative effect after instruct. What might be the reason?

**Comment by natolambert** [comment]
@jordane95 it could be different evaluation frameworks between `open-instruct` and the Llama team. This happens a lot :)

**Comment by jordane95** [comment]
> @jordane95 it could be different evaluation frameworks between `open-instruct` and the Llama team. This happens a lot :)

But you successfully reproduced the eval results of the base model...

**Comment by natolambert** [comment]
@jordane95 what do you mean? Can you be more specific, I think I misunderstood your question.

**Comment by jordane95** [comment]
> @jordane95 what do you mean? Can you be more specific, I think I misunderstood your question.

Let's look at the table below

| mmlu-5shot | llama65b | llama65b-instruct |
| :----: | :----: | :---: |
| llama paper | 63.4 | 68.9 |
| tulu paper | 63.3 | 62.4 |
| diff | 0.1 | 6.5 |

Looking at the diff, I bet it might not be the eval framework since the base model performs nearly the same, but there is a large gap in the instruct model performance.

**Comment by natolambert** [comment]
These are differences in eval setup. E.g. 0 vs 5 shot, and there are other settings. We've seen this before :)

Llama paper
<img width="330" alt="Screenshot 2024-06-29 at 10 11 03 AM" src="https://github.com/allenai/open-instruct/assets/10695622/62a1c499-28c9-445a-8ee4-2d108fc18c21">

Tulu 1 paper
<img width="582" alt="Screenshot 2024-06-29 at 10 11 35 AM" src="https://github.com/allenai/open-instruct/assets/10695622/38a9a8c9-11bd-4995-a112-73d0acd1a06f">

There's a difference between EM and MC. See this blog post for more details https://huggingface.co/blog/open-llm-leaderboard-mmlu

**Comment by hamishivi** [comment]
Hi, just to add some other notes: the llama-instruct model in the Llama 1 paper is trained purely on FLAN data, which is useful for MMLU, while for Tulu v1, the 65B model only saw ~100k FLAN samples (vs the full dataset size, which is something like 2 million datapoints). This may be why llama-instruct achieves much stronger MMLU - it is trained on much more of FLAN! Additionally, the tulu v2 mixture itself is much smaller than flan (~400k samples vs ~2 million).

We did use the same evaluation setup, as best we could, although we cannot rule out the possibility of small changes. One thing we did change is that we used 8bit quantization during MMLU evaluation for tulu v1, which may degrade scores slightly. However, I think the big shift from using only FLAN to a small mixture of it likely explains this discrepency.

**Comment by jordane95** [comment]
> Hi, just to add some other notes: the llama-instruct model in the Llama 1 paper is trained purely on FLAN data, which is useful for MMLU, while for Tulu v1, the 65B model only saw ~100k FLAN samples (vs the full dataset size, which is something like 2 million datapoints). This may be why llama-instruct achieves much stronger MMLU - it is trained on much more of FLAN! Additionally, the tulu v2 mixture itself is much smaller than flan (~400k samples vs ~2 million).
> 
> We did use the same evaluation setup, as best we could, although we cannot rule out the possibility of small changes. One thing we did change is that we used 8bit quantization during MMLU evaluation for tulu v1, which may degrade scores slightly. However, I think the big shift from using only FLAN to a small mixture of it likely explains this discrepency.

Hi @hamishivi , thanks for the reply. In fact, I have used the pure FLAN-v2 data released by https://huggingface.co/datasets/Open-Orca/FLAN to fine-tune llama. I observed the same trend as your paper stated. For MMLU 5-shot generation, there is an >10 points improvement at 7b size, but at 65b size, the score just doesn't change. That's why I'm so curious about the discrepancy with the results reported by llama paper. Maybe there is something wrong when training at large-scale?

**Comment by hamishivi** [comment]
Interesting! A few thoughts here:
- FLAN is quite unbalanced, and some subsets dominate its size. In the original implementation of FLAN, this isn't an issue because the code samples from the datasets in even-ish ways. Depending on how you are sampling, this might have a big impact.
- It doesn't seem like the Llama 1 paper gives details on hyperparameter choices, which might matter. Larger models often need a slightly lower LR, bigger batch sizes, etc. I've seen that even small tweaks in learning rates can matter for e.g. llama 3 finetunes.
- There may be other implementation details that matter: for example, if they did packing, how much gradient accumulation was used - these can also have effects, since using small batch sizes without gradient accumulation != using larger batch sizes depending on how the loss is calculated (since most implementations average over tokens within the minibatch, but gradient accumulation averages over minibatches as a whole).
- The trend of data giving smaller benefits as size increases matches most of my observations when finetuning - when the base model is already quite capable, it is hard to get any improvements from finetuning on generalist data, and tuning the data mixture, hyperparameters may matter more (while at the 7b size it is 'easy' for the model to learn improvements from the data).

**Comment by hamishivi** [comment]
@lucasliunju Hi, unfortunately we never trained llama 2 7b on FLAN alone, so I don't knew the results. Relevant 0-shot performances:
llama 1 7b (base): 31.5
llama 1 7b + FLAN: 45.4
Tulu 1 7b: 45.5
llama 2 7b (base): 41.8
Tulu 2 7b: 50.4
